tests/test_tape_recorder.py

```diff
@@ -1065,3 +1081,27 @@ def execute(self):
                                                       playback_function=lambda recording: Operation().execute())
         self._assert_playback_vs_recording(playback_result, result)
         wrapped.assert_called()
+
+    def test_record_and_playback_basic_operation_data_interception_copy_data(self):
```

Comment:
You are not checking in the recording the data is a copy and not original input
You should fetch last recording after execute and do recording.get_data and see you got a copy
So you need to change this from integer input to a dict input do you can actually check if it is the same instance

Reply:
the in memory save recording encodes it anyway, so it kind passes in any event. I can add a patch that checks that encode is called with 5.